com/win32com/client/__init__.py

```diff
@@ -269,7 +269,7 @@ def __get_disp_and_event_classes(dispatch):
     disp = Dispatch(dispatch)
 
     if disp.__class__.__dict__.get("CLSID"):
-        return disp.__class__
+        disp_class = disp.__class__
 
     # Eeek - no makepy support - try and build it.
     error_msg = "This COM object can not automate the makepy process - please run makepy manually for this object"
```

Comment:
it looks like we shouldn't be entering this block at all when `disp_class is not None`?

Reply:
oops, correct